app/static/js/keycodes.js

```diff
@@ -0,0 +1,144 @@
+var keyCodeLookup = {
+    "Pause / Break": 3,
```

Comment:
Since we exclusively look up keycodes character by character when translating a paste command, can we exclude keycodes for keys that are not representable in a paste? It doesn't seem like we need things like `"Pause / Break"` or even `"F1"` because I don't believe it's possible to paste any of those keys.

Reply:
Yeah, we can definitely trim down that lookup to only "useful" keys, i.e. ones you can actually type.

Reply:
I've been looking at the `keypress` listener to see whether that might get around the lookup table, but the main issue so far seems to be that using it instead of `keyup`/`keydown` is that `keypress` doesn't send through all key presses, only "characters" (https://javascriptinfo.com/view/5064780)

Reply:
I may be misinterpreting the dialog here, but would like to ask whether when pasting if each individual key is represented in the history box.  Personally, I'd like to simply see `<Paste>` or something to the effect in the history.  My use case is passwords; more often than not, I'd be using paste to paste in passwords (like BIOS password or something), and I'd rather not see it in the page's history.

Again, I may be mis-reading above.  Thanks!

Reply:
@rca - We can definitely refine this in the future. For the moment, we're looking to get the simplest possible working implementation. The easiest first implementation is to just translate the pasted text into a series of keystrokes since we already have handling for that without needing to change much server-side. But your use case makes sense. Once we get this merged in, we can look at doing the text-to-keystroke translation server-side so that the event that shows up in the history is just a "paste" event rather than the text itself.